store/postgres/src/connection_pool.rs

```diff
@@ -522,8 +568,11 @@ impl std::fmt::Debug for ErrorHandler {
 
 impl r2d2::HandleError<r2d2::Error> for ErrorHandler {
     fn handle_error(&self, error: r2d2::Error) {
-        self.1.inc();
-        error!(self.0, "Postgres connection error"; "error" => brief_error_msg(&error));
+        self.counter.inc();
+        if self.state_tracker.is_available() {
```

Comment:
This `handle_error` function, it really was never called when the database was unavailable?

I ask this because we would always log the error message from PostgreSQL it seems, so the connection should be working I beleive.

Reply:
Yes, it does get called when the database is unavailable. The only thing this change does is only log the error the first time the db becomes unavailable rather than spamming the logs with the same error message.

Reply:
Ohhh okay then.